Deploying Directus to Microsoft Azure Web Apps (#57)
directus-community/guest-authoring: directus-az/index.md

```bash
openssl rand -base64 32
```
This command uses OpenSSL to generate a random base64-encoded string of 32 characters. You can adjust the length as needed.

## Deploying Directus on Azure Web App
Before delving into the deployment of Directus on Azure Web App, it's essential to understand the seamless integration capabilities that Azure Web App offers. Azure Web App provides a fully managed platform for building, deploying, and scaling web apps. With its support for multiple programming languages, automated deployment, and streamlined scaling, Azure Web App simplifies the process of hosting web applications, making it an ideal choice for deploying Directus.

Similar to how our PostgreSQL Database was created, visit the Azure Marketplace and search for a Web App Resource, select the Web App resource highlighted below to create an Azure Web App.

![Azure web app](create-webapp.png)

> Within the Basics Configuration section: 
> - Select the resource group we created earlier.
> - Give your App instance a name.
> - Select **Docker Container** in the **Publish** settings. 
> - Choose your preferred operating system.
> - Select your preferred Region.
> - Select your preferred pricing plan.
> 
> ![Azure Web App Basic Configuration](web-configuration.png)
> 
> Click the **Next:Docker** button to configure the docker container settings for your web app using the docker-compose.yml we created earlier. Adjust the settings as seen below.
> 
> ![Docker congiguration settings](docker-configuration.png)
> 
> Click the **Next:Networking** button and select your preferred network settings.
> Finally, click on **Review + Create** and then **Create** to create your new Web App
> 
> Directus is successfully deployed and can be visited via the **Default domain** in the Azure web app. Leveraging the power of Directus' Backend as a Service (BaaS) solution, you can now benefit from ready-made REST and GraphQL APIs, seamless integrations, and superior scalability, empowering you to uncomplicate your backend operations with Directus.

Reviewer: ```suggestion
When creating a Web App, you will step through multiple configuration pages. Here are the important settings:

### Basics

- Subscription/Resource Group: select the same resource group we created and used earlier.
- Publish: Select "Docker Container". 

![Azure Web App Basic Configuration](web-configuration.png)

### Docker

Select Docker Compose and Docker Hub as the source for your app's configuration. Set the Docker Hub Access Type to Public and upload your `docker-compose.yml` file prepared earlier.

![Docker congiguration settings](docker-configuration.png)

Following the creation of the Web App Resource, Directus is now successfully deployed and can be visited via the default domain in the Azure Web App page. 
```

> ## Trouble Shooting Tips

Reviewer: ```suggestion
## Troubleshooting Tips
```

2. **Azure Web App Deployment Failures**

In case your Azure Web App deployment fails, consider the following:

- **Docker Image Compatibility:** Ensure that the Directus Docker image version specified in your `docker-compose.yml` file is compatible with the Azure Web App environment. Check for updates or use a different version if needed.

- **Resource Group Permissions:** Confirm that the Azure account used for deployment has the necessary permissions to create and manage resources within the specified resource group.

- **Docker Configuration Validation:** Validate your `docker-compose.yml` file for syntax errors or inconsistencies. Incorrect configurations can lead to deployment failures.
  
3. **Directus Interface Login Issues**

If you experience problems logging into the Directus interface:

- **Admin Credentials:** Ensure that the `ADMIN_EMAIL` and `ADMIN_PASSWORD` values in your `docker-compose.yml` file match the credentials you are using to log in.

- **Environment Variable Changes:** If you make changes to environment variables after the initial deployment, restart the Directus container to apply the new configurations.

## Summary

> Deploying Directus on Azure Web App provides a robust solution for content management, offering scalability and flexibility for your web development projects. By leveraging Docker and Azure services like Azure Database for PostgreSQL, you create a reliable and efficient environment for managing databases and content.
> 
> This tutorial has guided you through setting up a resource group, configuring Azure Database for PostgreSQL, and deploying Directus using Docker on Azure Web App. The integration of these technologies allows you to take advantage of the features offered by Directus while benefiting from the reliability and scalability of Azure services.

Reviewer: ```suggestion
This tutorial has guided you through setting up a resource group, configuring Azure Database for PostgreSQL, and deploying Directus using Docker on an Azure Web App.
```
